app/models/location.rb

```diff
@@ -245,7 +243,7 @@ def self.official_unknown
   # Get an instance of the Name that means "unknown".
   def self.unknown
     names_for_unknown.each do |name|
-      location = Location.find_by("name LIKE ?", name)
+      location = Location.find_by(Location[:name].matches(name))
       return location if location
     end
```

Comment:
1. Is there an **easy** Arel way to match one of a list? 
I.e., matches 'a' OR matches 'b' ...?
If so, then you wouldn't have to iterate over `names_for_unknown`.
2. @pellaea: Possible preexisting issue. Almost certainly for another PR.
Why `name LIKE ?` (and the Arel equivalent).
Does this risk returning a Location whose name merely includes the name of an unknown location?
(E.g., analogous to "[Black Earth Community Park, Wisconsin, USA](https://mushroomobserver.org/location/show_location/17558?q=1n0zv)")
Why do we allow "translations" of the unknown Location's name?

Reply:
@JoeCohen  it would be `matches_any(names_for_unknown)` but i believe we need a guard clause to be sure `names_for_unknown` is not empty

Reply:
@JoeCohen Just tried it locally. The difference between the results for 
```ruby
    Location.where(Location[:name].matches_any(names_for_unknown)).first
```
and going through the list in order, is that "Unknown" is first in the array `names_for_unknown` so the location "Unknown" will always be returned first in the `each` block. 

Hang on, I'm going to try a `find_in_set` instead of `matches_any` — the former checks the placement in the set.

Reply:
Hmm. Arel extensions find in set (`&`) didn't do it.
The array `names_for_unknown` in English returns
```ruby
["Unknown", "Earth", "World", "Worldwide", "Anywhere", "Everywhere"] 
```
We need a way to match that first one, the `each` block may be the best way.

Reply:
Should we do
```ruby
    Location.where(Location[:name].matches(names_for_unknown.first)).first
```

Reply:
Will this work:
```ruby
    location = Location.find_by(Location[:name].matches_any(names_for_unknown))
    return location if location
```

Reply:
According to the test, it works even without assigning it to the variable `location`.

Maybe `Earth` is the correct match here? If i try to force it to match `names_for_unknown.first` i.e. "Unknown" it returns `nil`.

Reply:
Committing your suggestion here, minus the variable.

Reply:
I have to take a break, but will look at it further in an hour or so.

Reply:
I don't understand how this is used, but:
- in master & in test mode, it returns the Location named "unknown", so it appears correct that it's returning the same thing in this branch;
- I think you should keep the guard clause and the error handling. 
`names_for_unknown` calls `:unknown_locations.l`.
At the moment, all languages  have "translations" for `unknown_locations`. But some random user could delete them, and I think we want to deal with that possibility.

Reply:
I've got no idea how this is used, either.  Has anyone checked where in the code this is called?  It may be out-dated and up for the chop block.

Reply:
No, it's definitely still used.  But I think it pre-dates the lock we put on "Earth".  How about we hard-code "Earth" instead.  Well, not *hard-code* hard-code, I mean make it a global constant, perhaps in config/consts.rb?  Would that help streamline this method?  Then it would just be: Location.where(name: MO.unknown_location_name) or the equivalent that returns just the first matching instance.

Reply:
Made it so.

Reply:
Okay, that blew up the tests. I added the constant to config/consts.rb:
```ruby
  config.unknown_location_name = "Earth"
```
This was the revised method I tried, with guard clause checking the constant:
```ruby
  def self.unknown
    raise("There is no \"unknown_location_name\" configured!") if
      MO.unknown_location_name.blank?
    Location.find_by(name: MO.unknown_location_name)
  end
```
Caused [many errors](https://github.com/MushroomObserver/mushroom-observer/runs/6586310302?check_suite_focus=true) in the tests.